android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -57,20 +62,25 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-        handleVisibility();
-
-        globalSearchAdapter = new GlobalSearchAdapter(results, this);
-        searchRecyclerView.setAdapter(globalSearchAdapter);
-        searchRecyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
-        searchRecyclerView.setLayoutManager(new LinearLayoutManager(this));
-
-        if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
-            searchText = savedInstanceState.getString(SEARCH);
-        }
+        setUpRecyclerView();
 
         if (Intent.ACTION_SEARCH.equals(getIntent().getAction())) {
             handleIntent(getIntent());
         }
+
+        searchViewModel = ViewModelProviders.of(this).get(SearchActivityViewModel.class);
+        searchText = searchViewModel.getSearchText();
+        if(searchViewModel.getSearchResults()!=null) {
```

Comment:
It's not the right way to do it. Infact even after doing this null check, the ```searches``` list could be null. The LiveData is just a wrapper around it. I would suggest you remove this null check, and instead of returning ```null```, return an empty list.

Reply:
Oh yeah that could be a problem. Sure I will change it 👍